engine/table/src/main/java/io/deephaven/engine/table/impl/ssms/DoubleSegmentedSortedMultiset.java

```diff
@@ -1341,7 +1341,7 @@ private static boolean geq(double lhs, double rhs) {
 
     private static boolean eq(double lhs, double rhs) {
         // region equality function
-        return lhs == rhs;
+        return doComparison(lhs, rhs) == 0;
```

Comment:
This change is the primary correction for https://github.com/deephaven/deephaven-core/issues/5820